### DISCUSSION TITLE
Fandango at Home (Vudu) Issue

### DISCUSSION THREAD

**Discussion by ghost** [aclap-dev/video-downloadhelper#1200] [Help and Q&A]
### Your question

Dear Video Downloader community,

I hope you all are doing well. So back in 2021, I bought the fourth season of Fargo on Vudu during a sale. I've gotten into a kick of slowly backing up every film and television purchase I've made, so I wanted to download the video files of the season for personal usage. I'm specifically using the Edge version since Vudu seems to only output 1080p through here. However, every time I attempt to download one episode, I get the following message:

"Sadly we failed at downloading that specific media. We try to support as many websites as possible, so it would help a lot if you could report that error (it's anonymous!)."

Now it's possible the there's no means of downloading it from this website or browser. And I'm not really interested in trying the other browsers I have since they only output Vudu at 480p. So, I figured I just share my issue as a last-ditch effort before I move on. If you have any insights I'm missing, I'm all ears. If not, totally understandable. Thank you, and have a good day.

### Addon & CoApp version and details

1. Using the latest version on Edge.
2. I've downloaded the latest CoApp version.
3. It says this message upon failure: Sadly we failed at downloading that specific media. We try to support as many websites as possible, so it would help a lot if you could report that error (it's anonymous!).

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

You should really just post what you get from the `Copy information to clipboard` button in the VDH settings.

Vudu is a paid site, right?  In that case, there isn't much anybody here can do to help you . . . unless a fellow subscriber to Vudu happens to stumble upon this discussion & offers some help.  You might post a series of screenshots.  This would be a simulation of letting us watch over your shoulder while you try to download something.

The error message has a `Report` button.  You should click that on just one of your error videos.  That does send a report to @paulrouget Paul (the developer) & he does analyze them.

**Reply by ghost** [reply]
Here is what the settings specifically say: 
version: 9.1.0.43
target: microsoft
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36 Edg/128.0.0.0
{
    "used_history_button": true,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "open_count_store": 14,
    "dable_attempts": {
        "__serializer_tag": "map",
        "__serializer_value": [
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "www.vudu.com"
                    },
                    {
                        "__serializer_tag": "array",
                        "__serializer_value": [
                            {
                                "__serializer_tag": "primitive",
                                "__serializer_value": 3
                            },
                            {
                                "__serializer_tag": "primitive",
                                "__serializer_value": 0
                            }
                        ]
                    }
                ]
            }
        ]
    },
    "dable_attempts_last": 1725140051278,
    "record_download_history": true
}

And here is the screenshot. Now I'm not the most code savvy person. So the clipboard reads like a foreign language to me. I also wanted to mention that when it initially happened, I did send a report. But I decided to also post it here to see if maybe someone who uses the add on had some insight. I appreciate any help, even if it ends up with me realizing I can't download my purchase haha.

![Screenshot 2024-09-01 015008](https://github.com/user-attachments/assets/850b80c8-8489-4b60-8383-22e556dc0c54)

**Reply by GrampaWildWilly** [reply]
I got as far as this page:

https://www.vudu.com/content/browse/details/Fargo-Season-4/1507184

It lists that particular episode of that particular season.  But if I want to look at the episode, I need to buy it.  It's only $2 but I'm not interested in the content.  When you click on the button in VDH to do the download, does it at least start to download the episode, or does it just fail right away?  I note that it's recognized as DASH by VDH.  That makes me think that it might be worth it to try the VDH beta, which is available only for Firefox.  You apparently don't have a VDH license for Edge so it's not like you would be losing access to your license to test this on Firefox.  Such a test would eliminate whether this is a solved bug.  Of course, if it is solved, the solution isn't available yet for Edge, only for Firefox in the VDH beta.  Which doesn't do you a whole lot of good.

Also, I see Vudu is apparently part of the NBC empire.  I spend a fair bit of my time in another corner of that empire: NBC Sports.  I watch golf tournament livestreams.  Those are protected by DRM.  It would not surprise me to learn that your content is also protected by DRM, which would mean downloading would be impossible.  Actually, what happens is VDH downloads the item (sometimes) & it looks like everything went fine.  But when you try to play it back, it's scrambled.  Determining whether something is protected by DRM is, so I'm told, not so easy on Edge (or Chrome).  But it's a piece of cake on Firefox.  To the left of the URL at the top of the browser window, Firefox puts up icons.  When you hover the mouse over each one, a tooltip hovers up.  When something is protected by DRM, Firefox puts up an icon whose tooltip mentions DRM.  I see in your screen shot that Edge has put up one icon in the area I'm talking about.  This would be another incentive to at least look at your page in Firefox, to determine whether there's DRM protection on the item.

**Comment by Mattmjs** [comment]
Is it any surprise that a site offering movies and tv shows that you either buy or rent uses DRM protection.
I view a trailer on there, same failure to download it but also Firefox indicates I need to enable DRM as in the screenshot below.
I'm assume to watch the content you've paid for requires DRM running to watch it. Firefox will show a DRM icon near the padlock if
that's the case. But Chrome or Edge won't give a clue on that.

![vudu](https://github.com/user-attachments/assets/d9bd69f6-510f-406d-bc10-fe3c3bb916c0)
